include/eve/detail/function/simd/riscv/slice.hpp

```diff
@@ -0,0 +1,187 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/category.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/module/core/regular/simd_cast.hpp>
+
+// Return first or second part of vector.
+
+namespace eve::detail
+{
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE wide<T, typename N::split_type>
+                riscv_lmul_trunc(wide<T, N> a) noexcept
```

Comment:
Can't simd_cast do this now?

Reply:
In final result we will call the same consequence of intrinsics, as `simd_cast` will call slice, that call trunc. For now I expect this should be used only there, and not in common code.

For common code, I suppose we can be better: there I have restriction that result `lmul` at maximum should be twice smaller that original `lmul`, but for `simd_cast` there is no such restriction. And we can just cast with one intrinsic to needed lmul (but this will be too big function to review with this patch).

Reply:
I have a problem with duplicating the same intrinsic selection logic twice. Either do it in simd_cast or in slice, don't do it twice.

Reply:
> same intrinsic selection logic twice

For now it is used only here. Of course, when I will implement a more general algorithm, I will move this functionality there. I can move this to `simd_cast` now, but my concern is that it will result in huge block of untested code (as tests for `simd_cast` not in `unit.arch`).

Added TODO with this comment for now.